pyproject.toml

```diff
@@ -125,6 +125,7 @@ module = [
   "netCDF4.*",
   "netcdftime.*",
   "opt_einsum.*",
+  "pandas.*",
```

Comment:
Is this necessary? IIUC it means we miss out on any pandas types...

Reply:
Without this I get a large number of `error: Library stubs not installed for "pandas"  [import-untyped]`.

Reply:
Hmmm, does `mypy` pass on the main branch for you? If not, it might be worth trying something like `mypy --install-types`.

If it passes on main but requires this addition for this branch's proposed changes, we should work out why...

Reply:
ah good point, this fixed the issue

Reply:
I think a lot of these type: ignores were because of pandas stubs.
So you will have to revert lots of those when using pandas stubs.